### ISSUE TITLE
thoughts on colorful test output

### ISSUE THREAD

**Issue by softprops** [bazelbuild/rules_scala#255]
At the moment it seems likely any streamed test output is [hardcoded](https://github.com/bazelbuild/rules_scala/blob/master/scala/scala.bzl#L685) to suppress color ( -W means without color according to the [Runner docs](http://www.scalatest.org/user_guide/using_the_runner) ). For engineers that are coming from sbt and onto baze, color is something that is missed. It also holds practical value to have test failures visually jump out at you in red. What are your thoughts on a patch that would make color and option available to end users?

**Comment by johnynek** [comment]
I think having an option for scala_test for this (and short stacks for
instance) would be a great idea.

**Comment by softprops** [comment]
Good to hear. I was just taking the temperature before I sketched out a branch. I may not get to this immediately but will likely circle back at some point.

**Comment by ittaiz** [comment]
Doesn't sound like specs2 and JUnit are a priority for you but I'll just
mention I hope to dumb it down to the macro level of calling scala library
and java_test. Need to see when I'll get to it

**Comment by softprops** [comment]
I think the two, or three different test runners can work this in as separate efforts for the different rule interfaces. We mainly only use scala_test so that's the one I'd likely be targeting. Our engineers really like its colored output in sbt and we're trying to make the transition a pleasant one.

**Comment by softprops** [comment]
I've got a branch for this but am unsure what a good test for this looks like. would it be okay for open a prelim or to get some initial feedback on what a test might look like?

**Comment by ittaiz** [comment]
Definitely.
Though I have to warn you that I'm working on a big diff with moving from
manual javac to java_common.compile and adding `strict java deps` support
to java.

**Comment by softprops** [comment]
Rendered https://github.com/bazelbuild/rules_scala/pull/268

**Comment by ittaiz** [comment]
I agree. I've split the change to small logical commits which are
reviewable.
The biggest one and the one containing 80% of the logic is 75 lines
addition and about tha in deletions.
It's big because it touches all existing code paths and reasoning about it
in our god file bzl is hard.

We want to do a major restructuring of it to break apart the big ball of
mud but we're currently tied up. Of course brainstorm the solution together.

**Comment by softprops** [comment]
Just looking to guesstimate if I can factor in a upgrade of rules scala in an upcoming sprint at work. For this particular change, it's pretty small but wasn't sure what good tests would look like. I'm happy to wait on a larger pull to make its way through but do we have an eta on when that would happen? On my end, my work task just looks like "improve the bazel testing experience". I think I can accomplish alot with few lines of change but I want to stick as close to head of rules scala as I can.

**Comment by ittaiz** [comment]
I think I'll open the PR today BUT:
1. I suspect there are two missing bits (I might have broken IntelliJ
support and some meta aspects)
2. I'm on vacation Tuesday-Friday
Given both of the above I estimate this will be merged Monday next week

**Comment by johnynek** [comment]
Can we close this out now?

**Comment by softprops** [comment]
Case closed!
